src/app/components/Item/ItemTableRow.jsx

```diff
@@ -39,28 +38,7 @@ class ItemTableRow extends React.Component {
     Actions.updateLoadingStatus(true);
 
     trackDiscovery('Item Request', gaLabel);
-    ajaxCall(`${appConfig.baseUrl}/api/hold/request/${bibId}-${item.id}`,
-      (response) => {
-        Actions.updateBib(response.data.bib);
-        Actions.updateDeliveryLocations(response.data.deliveryLocations);
-        Actions.updateIsEddRequestable(response.data.isEddRequestable);
-        setTimeout(() => {
-          Actions.updateLoadingStatus(false);
-          this.context.router.push(`${appConfig.baseUrl}/hold/request/${bibId}-${item.id}`);
-        }, 500);
-      },
-      (error) => {
-        setTimeout(() => {
-          Actions.updateLoadingStatus(false);
-        }, 500);
-
-        // eslint-disable-next-line no-console
-        console.error(
-          `Error attemping to make an ajax request to fetch an item on ${page}`,
-          error,
-        );
-      },
-    );
+    this.context.router.push(`${appConfig.baseUrl}/hold/request/${bibId}-${item.id}`);
```

Comment:
So this essentially changes the URL to the create-hold-request route, which will trigger a redirect to login.nypl [here](https://github.com/NYPL-discovery/discovery-front-end/blob/c28c463d3f56129aeea7d6fc498f2393f2be294e/src/app/components/HoldRequest/HoldRequest.jsx#L150) .. if the patron is not logged in. Otherwise, the api route (the ajax call you've removed above) will be auto fetched on-load?

Reply:
I wonder if that line of code Paul linked to can also use router context and if that will be less buggy. This approach is pretty old according to the blame.

Not super significant, but since I did the poking around, it's from [this PR](https://github.com/NYPL-discovery/discovery-front-end/pull/41) four years ago. The olden days!

Reply:
Yes. Unfortunately the api route will be auto fetched on load in both cases, even though it will definitely error if the user is not logged in, but it is a silent error.

Reply:
This logic is also in `src/app/components/ElectronicDelivery/ElectronicDelivery` by the way

Reply:
A silent error seems fine. Correct even. Ideally we wouldn't attempt to make the ajax call if we don't expect it to return anything (i.e. because the patron is not logged in), but that doesn't seem like the worst thing.

The use of `window.location.href` over `router.push` feels okay to me? Only because it's a different domain. Are there benefits to using `router.push` when navigating away from the app to an entirely different domain?

Reply:
It's actually `window.location.replace`. Not sure if that effects your opinion at all. I guess the use of `replace` makes sense, because otherwise the back button would go to the hold request page url again and redirect again. There is `router.replace` as well I believe. I think I just prefer it for consistency. I don't think there is any reason you cannot link to a different domain using `router.push`.